app/models/subscribe_link.rb

```diff
@@ -0,0 +1,180 @@
+    "snipd" => "https://share.snipd.com/show/${uniquePlatformID}",
+    "sonnet" => "https://sonnet.fm/p/${appleID}",
+    "steno" => "https://steno.fm/show/${podcastGUID}",
+    "truefans" => "https://truefans.fm/${podcastGUID}",
+    "tunein" => "https://tunein.com/podcasts/${uniquePlatformID}"
+  }
+
+  PLATFORM_ICONS = {
+    "apple" => "apple",
+    "spotify" => "spotify",
+    "overcast" => "overcast",
+    "pocketcasts" => "pocketcasts",
+    "youtube" => "youtube",
+    "youtube_feed" => "youtube",
+    "amazon" => "amazon",
+    "antenna" => "antenna",
+    "anytime" => "anytime",
+    "apollo" => "apollo",
+    "breez" => "breez",
+    "castamatic" => "castamatic",
+    "castbox" => "castbox",
+    "castro" => "castro",
+    "curiocaster" => "curiocaster",
+    "deezer" => "deezer",
+    "fountain" => "fountain",
+    "global" => "global",
+    "goodpods" => "goodpods",
+    "gpodder" => "gpodder",
+    "hark" => "hark",
+    "iheart" => "iheart",
+    "lnbeats" => "lnbeats",
+    "luminary" => "luminary",
+    "moon" => "moon",
+    "pandora" => "pandora",
+    "player" => "player",
+    "podbean" => "podbean",
+    "podaddict" => "podaddict",
+    "podapp" => "podapp",
+    "podguru_apple" => "podguru",
+    "podguru_feed" => "podguru",
+    "podrepublic" => "podrepublic",
+    "podfriend" => "podfriend",
+    "podlp_apple" => "podlp",
+    "podlp_guid" => "podlp",
+    "podstation" => "podstation",
+    "podurama" => "podurama",
+    "podverse" => "podverse",
+    "snipd" => "snipd",
+    "sonnet" => "sonnet",
+    "steno" => "steno",
+    "truefans" => "truefans",
+    "tunein" => "tunein"
+  }
+
+  APPLE_PLATFORMS = %w[apple overcast pocketcasts castbox castro goodpods moon podbean podguru_apple podrepublic podfriend podlp_apple podurama sonnet]
+
+  UNIQUE_PLATFORMS = %w[spotify youtube amazon deezer global hark iheart luminary pandora podapp snipd tunein]
+
+  FEED_PLATFORMS = %w[youtube_feed antenna anytime apollo breez gpodder player podaddict podguru_feed podstation]
+
+  GUID_PLATFORMS = %w[castamatic lnbeats podlp_guid steno truefans]
+
+  POD_INDEX_PLATFORMS = %w[curiocaster fountain podverse]
+
+  SLUG_PLATFORMS = %w[luminary pandora podapp podurama]
+
+  belongs_to :podcast, -> { with_deleted }, optional: true, touch: true
+
+  scope :with_apple_id, -> { where(platform: APPLE_PLATFORMS) }
+
+  validates :platform, presence: true, inclusion: {in: PLATFORMS}
+  validates :external_id, presence: true
+
+  def uses_apple_id?
+    APPLE_PLATFORMS.include?(platform)
+  end
+
+  def uses_unique_id?
+    UNIQUE_PLATFORMS.include?(platform)
+  end
+
+  def uses_feed_url?
+    FEED_PLATFORMS.include?(platform)
+  end
+
+  def uses_podcast_guid?
+    GUID_PLATFORMS.include?(platform)
+  end
+
+  def uses_pod_index_id?
+    POD_INDEX_PLATFORMS.include?(platform)
+  end
+
+  def uses_slug?
+    SLUG_PLATFORMS.include?(platform)
+  end
+
+  def icon
+    PLATFORM_ICONS[platform]
+  end
+
+  def href
```

Comment:
more preparation for the json PR to follow this one, but essentially i take the href mapping above and substitute in the `external_id` where applicable, depending on what that href uses.